serde/src/json/de.rs

```diff
@@ -267,8 +267,16 @@ impl<Iter> Deserializer<Iter>
         while !self.eof() {
             match self.ch_or_null() {
                 c @ b'0' ... b'9' => {
-                    exp *= 10;
-                    exp += (c as i32) - (b'0' as i32);
+                    macro_rules! try_or_invalid {
```

Comment:
Could you pull this macro out to the top level since I also use it for `parse_decimal`?